admin/tool/policy/lib.php

```diff
@@ -74,16 +75,16 @@ function tool_policy_before_standard_html_head()
  * Hooks redirections to digital minor validation and policy acceptance pages before sign up.
  */
 function tool_policy_pre_signup_requests() {
+    global $SESSION;
 
-    if (!isset($SESSION->minor)) {  // Digital minor check hasn't been done.
+    if (!isset($SESSION->tool_policy->minor)) {  // Digital minor check hasn't been done.
```

Comment:
Should not this be replaced by some `validateminor_helper` method, too? So that the SESSION storage would be completely encapsulated by that class.